### PR TITLE
Add removeTechControlsListener call to add method

### LINKED ISSUE #2504: Since 5.0.0-rc60: Pause/Play functionality on click doesn't work any more
Since the update to 5.0.0-rc60, I cannot play/pause the video by clicking somewhere on the video-element any more. Used to work, now it doesn't on chrome, firefox and safari on mac. The issue is independent of the used media format, fails on both mp4 and webm.

### PR DESCRIPTION
**Pull request by gkatsev**
Fixes #2504.

### CODE REVIEW

**Comment by gkatsev** [comment]
This fixes the immediate issue. Looking into whether we can simplify when `addTechControlsListener` gets called.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 97f9ddbbfe978d47c6a109b05fbf0de82d369fe1
Build details: https://travis-ci.org/pam/video.js/builds/77048645

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 97f9ddbbfe978d47c6a109b05fbf0de82d369fe1
Build details: https://travis-ci.org/pam/video.js/builds/77048645

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by heff** [comment]
Apparently re-running savage tests in travis doesn't work.

@pam retry

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: c97a5259b9fadf0231ee3f0fd8b3d89c64ce0464
Build details: https://travis-ci.org/pam/video.js/builds/77060173

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
Failed with `vjs_getProperty`.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: b80e2b835401dd6d12c8a47d331e95b8cc6b8eed
Build details: https://travis-ci.org/pam/video.js/builds/77061751

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
@pam retry

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: b80e2b835401dd6d12c8a47d331e95b8cc6b8eed
Build details: https://travis-ci.org/pam/video.js/builds/77068616

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: src/js/player.js [hunk 1]
```diff
@@ -1939,7 +1942,6 @@ class Player extends Component {
       if (this.usingNativeControls_ !== bool) {
```
Comment (dmlap) [inline src/js/player.js]: Personally, I would invert this conditional and return early to get rid of a level of nesting in this function. Not worth blocking this PR, though.

**Comment by dmlap** [comment]
Do we have sufficient test coverage for this already?

**Comment by gkatsev** [comment]
I'll take a look at what tests we have for this.

**Comment by gkatsev** [comment]
I added a test.

File: src/js/player.js (hunk 1 above)
Comment (heff) [inline src/js/player.js]: Yeah, and you could do the same with the `bool !== undefined` above this.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: c930090dfcd82a87665f58e7bfe9e8dfcf8de673
Build details: https://travis-ci.org/pam/video.js/builds/77201275

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: test/unit/player.test.js [hunk 2]
```diff
@@ -415,6 +415,32 @@ test('should allow for tracking when native controls are used', function(){
...
+
+  player.controls(true);
+
+  equal(listeners, 0, 'addTechControlsListeners should not have gotten called yet');
+
+  player.usingNativeControls(false);
+  player.controls(false);
+
+  player.controls(true);
+  equal(listeners, 1, 'addTechControlsListeners should have gotten called once')
```
Comment (heff) [inline test/unit/player.test.js]: semicolon

File: test/unit/player.test.js [hunk 3]
```diff
@@ -415,6 +415,32 @@ test('should allow for tracking when native controls are used', function(){
   player.dispose();
 });
 
+test('make sure that controls listeners do not get added too many times', function(){
+  var player = TestHelpers.makePlayer({});
+  var listeners = 0;
+
+  player.addTechControlsListeners = function() {
+    listeners++;
+  }
```
Comment (heff) [inline test/unit/player.test.js]: semicolon

Comment (gkatsev) [inline test/unit/player.test.js]: d'oh.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 8276fd9e8fb3fcef71d49349ab95a0b9cb83c9fe
Build details: https://travis-ci.org/pam/video.js/builds/77202993

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
@pam retry

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 8276fd9e8fb3fcef71d49349ab95a0b9cb83c9fe
Build details: https://travis-ci.org/pam/video.js/builds/77213284

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
The errors are the `vjs_getProperty` issue which are coming from... _somewhere_... and handled elsewhere (#2515 or #2517).

**Comment by heff** [comment]
Assuming we've tested this on the platforms this was reported on, lgtm.

**Comment by gkatsev** [comment]
I've tested it locally on ipads, android native browser, chrome, firefox, safari

**Pull request closed without merging**